### PR TITLE
[docs] refactor datetime docs to focus on React usage

### LINKED ISSUE #2542: Documentation problem; DateRangeInput does not show required props
the line example for [DateRangeInput](http://blueprintjs.com/docs/v2/#datetime/daterangeinput) does not have the two required props, `formatDate` and `parseDate`. Trying to use that code will result in errors.
I'd be very happy to just fix this myself, but I'm not actually sure where in the repo the documentation lives to submit a pr.

### PR DESCRIPTION
**Pull request by giladgray**
fixes #2542 

refactor all datetime component docs a la #2574 (which this PR is targeting).

### CODE REVIEW

File: packages/datetime/src/daterangeinput.md [hunk 1]
```diff
@@ -8,23 +8,28 @@ Use this component in forms where the user must enter a date range.
...
 
-```
+```tsx
 import { DateRangeInput } from "@blueprintjs/datetime";
 
-<DateRangeInput value={[this.state.startDate, this.state.endDate]} onChange={this.handleChange} />
+<DateRangeInput
+    formatDate={date => date.toLocaleString()}
+    onChange={this.handleRangeChange}
+    parseDate={str => new Date(str)}
```
Comment (giladgray) [inline packages/datetime/src/daterangeinput.md]: added required props for #2542

File: packages/datetime/src/dateinput.md [hunk 2]
```diff
@@ -51,11 +48,17 @@ function getMomentFormatter(format: string): IDateFormatProps {
 <DateInput {...getMomentFormatter("LL")} locale="de" />
 ```
 
+@## Props
```
Comment (benjdj6) [inline packages/datetime/src/dateinput.md]: Write out the word "properties" for professionalism.

File: packages/datetime/src/dateinput.md [hunk 3]
```diff
@@ -51,11 +48,17 @@ function getMomentFormatter(format: string): IDateFormatProps {
...
+
+Use the `onChange` function to listen for changes to the selected date. Use
+`onError` to listen for invalid entered dates.
 
-@## JavaScript API
+You can control the selected date by setting the `value` prop, or use the
```
Comment (benjdj6) [inline packages/datetime/src/dateinput.md]: prop -> property

File: packages/datetime/src/datepicker.md [hunk 4]
```diff
@@ -2,43 +2,22 @@
...
-in the [**react-day-picker** documentation](http://www.gpbl.org/react-day-picker/index.html).
-
-@interface IDatePickerProps
+@## Modifiers
 
-@## Using modifiers
-
-You can use the `modifiers` prop to conditionally apply styles to days. Modifiers are documented in
-full in the [**react-day-picker** documentation](http://react-day-picker.js.org/docs/matching-days).
+You can use the `modifiers` prop to conditionally apply styles to days.
```
Comment (benjdj6) [inline packages/datetime/src/datepicker.md]: prop -> property

File: packages/datetime/src/datepicker.md [hunk 5]
```diff
@@ -2,43 +2,22 @@
...
+Modifiers are a react-day-picker concept and are documented in full in the
+[**react-day-picker** documentation](http://react-day-picker.js.org/docs/matching-days).
 
 The example below creates a `DatePicker` that prevents the user from selecting any Sundays,
 by using the component in controlled mode and with the `modifiers` prop:
```
Comment (benjdj6) [inline packages/datetime/src/datepicker.md]: prop -> property

File: packages/datetime/src/datepicker.md [hunk 6]
```diff
@@ -71,3 +50,16 @@ function isSunday(date: Date) {
     return date.getDay() === 0;
 }
 ```
+
+@## Props
```
Comment (benjdj6) [inline packages/datetime/src/datepicker.md]: Props -> properties

File: packages/datetime/src/datepicker.md [hunk 7]
```diff
@@ -71,3 +50,16 @@ function isSunday(date: Date) {
...
+
+`DatePicker` supports both controlled and uncontrolled usage. You can control
+the selected day by setting the `value` prop, or use the component in
```
Comment (benjdj6) [inline packages/datetime/src/datepicker.md]: prop -> property

File: packages/datetime/src/datepicker.md [hunk 8]
```diff
@@ -71,3 +50,16 @@ function isSunday(date: Date) {
...
+uncontrolled mode and specify an initial day by setting `defaultValue`. Use the
+`onChange` prop to listen for changes to the selected day.
```
Comment (benjdj6) [inline packages/datetime/src/datepicker.md]: prop -> property

File: packages/datetime/src/datepicker.md [hunk 9]
```diff
@@ -71,3 +50,16 @@ function isSunday(date: Date) {
...
+
+Some props are managed by the `DatePicker` component, while others are passed
```
Comment (benjdj6) [inline packages/datetime/src/datepicker.md]: prop -> property

File: packages/datetime/src/datepicker.md [hunk 10]
```diff
@@ -71,3 +50,16 @@ function isSunday(date: Date) {
...
+to the **react-day-picker** library. These passed props are documented in full
```
Comment (benjdj6) [inline packages/datetime/src/datepicker.md]: prop -> property

File: packages/datetime/src/daterangeinput.md [hunk 11]
```diff
@@ -8,23 +8,28 @@ Use this component in forms where the user must enter a date range.
 
 @reactExample DateRangeInputExample
 
-@## JavaScript API
+@## Props
```
Comment (benjdj6) [inline packages/datetime/src/daterangeinput.md]: prop -> property

File: packages/datetime/src/daterangeinput.md [hunk 12]
```diff
@@ -8,23 +8,28 @@ Use this component in forms where the user must enter a date range.
...
 
-The `DateRangeInput` component is available in the __@blueprintjs/datetime__ package.
-Make sure to review the [getting started docs for installation info](#blueprint/getting-started).
+Use the `onChange` function to listen for changes to the selected date. Use
+`onError` to listen for invalid entered dates or date ranges.
 
-Use the `onChange` function to listen for changes to the selected date. Use `onError` to listen for
-invalid entered dates or date ranges.
+You can control the selected date by setting the `value` prop, or use the
```
Comment (benjdj6) [inline packages/datetime/src/daterangeinput.md]: prop -> property

File: packages/datetime/src/daterangepicker.md [hunk 13]
```diff
@@ -3,23 +3,24 @@
...
+`DateRangePicker` uses the `DateRange` type across its API. This is an alias for
+the tuple type `[Date, Date]`.
+
+Semantically:
+* `[null, null]` represents an empty selection.
+* `[someDate, null]` represents a date range where a single endpoint is known.
+* `[someDate, someOtherDate]` represents a full date range where both endpoints are known.
 
-@## JavaScript API
+@## Props
```
Comment (benjdj6) [inline packages/datetime/src/daterangepicker.md]: prop -> property

File: packages/datetime/src/daterangepicker.md [hunk 14]
```diff
@@ -3,23 +3,24 @@
...
 
-The `DateRangePicker` component is available in the __@blueprintjs/datetime__ package.
-Make sure to review the [getting started docs for installation info](#blueprint/getting-started).
+Use the `onChange` prop to listen for changes to the set date range. You can
```
Comment (benjdj6) [inline packages/datetime/src/daterangepicker.md]: prop -> property

File: packages/datetime/src/datetimepicker.md [hunk 15]
```diff
@@ -1,22 +1,20 @@
...
-
-You can pass props to the inner `DatePicker` and `TimePicker` components using
-`datePickerProps` and `timePickerProps`, respectively.
+`DateTimePicker` composes a [`DatePicker`](#datetime/datepicker)
+and a [`TimePicker`](#datetime/timepicker) into one container.
 
 @reactExample DateTimePickerExample
 
-@## JavaScript API
+@## Props
```
Comment (benjdj6) [inline packages/datetime/src/datetimepicker.md]: prop -> property

File: packages/datetime/src/datetimepicker.md [hunk 16]
```diff
@@ -1,22 +1,20 @@
...
 
-The `DateTimePicker` component is available in the __@blueprintjs/datetime__ package.
-Make sure to review the [getting started docs for installation info](#blueprint/getting-started).
+Use the `onChange` prop to listen for changes to the selected date and time. You
```
Comment (benjdj6) [inline packages/datetime/src/datetimepicker.md]: prop -> property

File: packages/datetime/src/datetimepicker.md [hunk 17]
```diff
@@ -1,22 +1,20 @@
...
+can control the selected date and time by setting the `value` prop, or use the
```
Comment (benjdj6) [inline packages/datetime/src/datetimepicker.md]: prop -> property

File: packages/datetime/src/datetimepicker.md [hunk 18]
```diff
@@ -1,22 +1,20 @@
...
+component in uncontrolled mode and specify an initial day by setting
+`defaultValue`. (If `defaultValue` is not set, the current date and time is used
+as the default.)
+
+You can pass props to the inner `DatePicker` and `TimePicker` components using
```
Comment (benjdj6) [inline packages/datetime/src/datetimepicker.md]: prop -> property

File: packages/datetime/src/timepicker.md [hunk 19]
```diff
@@ -2,22 +2,17 @@
...
-
 `TimePicker` has no direct localization support. You should handle localization directly in your
 application if needed.
 
-`TimePicker` uses `Date` objects across its API but ignores their year, month, and day fields.
-
 @reactExample TimePickerExample
 
-@## JavaScript API
+@## Props
```
Comment (benjdj6) [inline packages/datetime/src/timepicker.md]: prop -> property

File: packages/datetime/src/timepicker.md [hunk 20]
```diff
@@ -2,22 +2,17 @@
...
+
+Use the `onChange` prop to listen for changes to the set time. You can control the selected time by
```
Comment (benjdj6) [inline packages/datetime/src/timepicker.md]: prop -> property

File: packages/datetime/src/timepicker.md [hunk 21]
```diff
@@ -2,22 +2,17 @@
...
+setting the `value` prop, or use the component in uncontrolled mode and specify an initial time by
```
Comment (benjdj6) [inline packages/datetime/src/timepicker.md]: prop -> property

File: packages/timezone/src/components/timezone-picker/timezone-picker.md [hunk 22]
```diff
@@ -7,7 +7,7 @@ timezone metadata.
 
 @reactExample TimezonePickerExample
 
-@## JavaScript API
+@## Props
```
Comment (benjdj6) [inline packages/timezone/src/components/timezone-picker/timezone-picker.md]: prop -> property

File: packages/datetime/src/dateinput.md (hunk 2 above)
Comment (giladgray) [reply packages/datetime/src/dateinput.md]: "props" is actually a react convention, and they're never called "properties" (to distinguish from JS objects)

Comment (giladgray) [reply packages/datetime/src/dateinput.md]: see React's own docs: https://reactjs.org/docs/components-and-props.html

**Pull request merged**